### ISSUE TITLE
Restarting a promoted build loses target environment

### ISSUE THREAD

**Issue by JeffDownie** [drone/drone#2748]
When I try to restart a build that has been run as the result of a promotion, drone seems to trigger a new promotion build, but with the variable set to empty - for example, if I run

`drone build promote my/repository 123 production`

It works fine, but when I hit the restart button in the UI for this build, it does something roughly the equivalent of

`drone build promote my/repository 123`

(after the 123 intentionally left blank)

**Comment by bradrydzewski** [comment]
this can be fixed by modifying the following:
https://github.com/drone/drone/blob/d06e688e8aac6ae41f5b2105cd085105300b24d6/handler/api/repos/builds/promote.go#L75

```diff
			Sender:       prev.Sender,
			Params:       map[string]string{},
		}

+		for k, v := range prev.Params {
+			hook.Params[k] = v
+		}
```

**Comment by bradrydzewski** [comment]
fixed by c7788c45872bde1ec9476d113c5bb07babf29ce9
